internal/services/dataprotection/data_protection_backup_policy_mysql_flexible_server_resource.go

```diff
@@ -0,0 +1,697 @@
+	Duration      string `tfschema:"duration"`
+}
+
+type BackupPolicyMySQLFlexibleServerRetentionRule struct {
+	Name      string                                     `tfschema:"name"`
+	Criteria  []BackupPolicyMySQLFlexibleServerCriteria  `tfschema:"criteria"`
+	LifeCycle []BackupPolicyMySQLFlexibleServerLifeCycle `tfschema:"life_cycle"`
+	Priority  int64                                      `tfschema:"priority"`
+}
+
+type BackupPolicyMySQLFlexibleServerCriteria struct {
+	AbsoluteCriteria     string   `tfschema:"absolute_criteria"`
+	DaysOfWeek           []string `tfschema:"days_of_week"`
+	MonthsOfYear         []string `tfschema:"months_of_year"`
+	ScheduledBackupTimes []string `tfschema:"scheduled_backup_times"`
+	WeeksOfMonth         []string `tfschema:"weeks_of_month"`
+}
+
+type DataProtectionBackupPolicyMySQLFlexibleServerResource struct{}
+
+var _ sdk.Resource = DataProtectionBackupPolicyMySQLFlexibleServerResource{}
+
+func (r DataProtectionBackupPolicyMySQLFlexibleServerResource) ResourceType() string {
+	return "azurerm_data_protection_backup_policy_mysql_flexible_server"
+}
+
+func (r DataProtectionBackupPolicyMySQLFlexibleServerResource) ModelObject() interface{} {
+	return &BackupPolicyMySQLFlexibleServerModel{}
+}
+
+func (r DataProtectionBackupPolicyMySQLFlexibleServerResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return backuppolicies.ValidateBackupPolicyID
+}
+
+func (r DataProtectionBackupPolicyMySQLFlexibleServerResource) Arguments() map[string]*pluginsdk.Schema {
+	arguments := map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.BackupPolicyMySQLFlexibleServerName,
+		},
+
+		"vault_id": commonschema.ResourceIDReferenceRequiredForceNew(pointer.To(backuppolicies.BackupVaultId{})),
+
+		"backup_repeating_time_intervals": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			ForceNew: true,
+			MinItems: 1,
+			Elem: &pluginsdk.Schema{
+				Type:         pluginsdk.TypeString,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+
+		"default_retention_rule": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			ForceNew: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"life_cycle": {
+						Type:     pluginsdk.TypeList,
+						Required: true,
+						ForceNew: true,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"data_store_type": {
+									Type:     pluginsdk.TypeString,
+									Required: true,
+									ForceNew: true,
+									ValidateFunc: validation.StringInSlice([]string{
+										// Confirmed with the service team that current possible value only support `VaultStore`.
+										// However, considering that `ArchiveStore` will be supported in the future, it would be exposed for user specification.
+										string(backuppolicies.DataStoreTypesVaultStore),
+									}, false),
```

Comment:
as there is only 1 possible value, could we make this optional with it as the default?

Reply:
Based on below reasons. I suggest aligning with other the existing data protection backup policies.

1. When new value is introduced in the future, it's very likely that the default value is no longer applicable and it may introduce the breaking change.
2. TF has supported many [data protection backup policies] (https://github.com/hashicorp/terraform-provider-azurerm/tree/main/internal/services/dataprotection). They are almost all the same. I feel it should be aligned with others.

Reply:
it is not a breaking change if a new value is added, as it would not need to be changed to required. however as all the other resources have it as required lets just keep it consistent